File: _c++/symulacja_organizmow/Organizmy/Zwierzeta/Owca.cpp
//
//  Owca.cpp
//  WirtualnySwiat
//
//  Created by Michal Tarnacki on 10/04/2022.
//

#include "Owca.hpp"

Owca::Owca(Swiat &swiat):Zwierze(swiat){
    SetInicjatywa();
    SetSila();
};
char Owca::Rysowanie() const{
    return 'O';
}
void Owca::SetInicjatywa(){
    this->inicjatywa=4;
}
void Owca::SetSila(){
    this->sila=4;
}
Organizm* Owca::Urodz(){
    return new Owca(this->swiat);
}
bool Owca::MozeBycPartnerem(Organizm& organizm){
    if(dynamic_cast<Owca*>(&organizm))
       return true;
    else
       return false;
}

File: _java/symulacja_organizmow/src/Symulacja/Zwierzeta/Owca.java
package Symulacja.Zwierzeta;
import Symulacja.*;

import java.awt.*;

public class Owca extends Zwierze {
    protected void SetInicjatywa(){
        this.inicjatywa=4;
    };
    protected void SetSila(){
        this.sila=4;
    };
    public Owca(Swiat swiat){
        super(swiat);
        SetInicjatywa();
        SetSila();
    };
    public char Rysowanie(){
        return 'O';
    };
    public Organizm Urodz() {
        return new Owca(this.swiat);
    };
    public boolean MozeBycPartnerem(Organizm organizm){
        if(organizm instanceof Owca)
        return true;
    else
        return false;
    };
    @Override
    public Color GetColor(){
        return new Color(0,0,100);
    };
}
